lms/static/js/instructor_dashboard/ecommerce.js

```diff
@@ -52,5 +57,139 @@ var edx = edx || {};
              }
            });
         });
+
+        String.prototype.capitalizeFirstLetter = function() {
+            return this.charAt(0).toUpperCase() + this.slice(1);
+        };
+        lookup_button.click(function () {
+            registration_code_status_form_error.attr('style', 'display: none');
+
+            lookup_button.attr('disabled', true);
+            var url = $(this).data('endpoint');
+            var lookup_registration_code = $('#set_regcode_status_form input[name="regcode_code"]').val();
+            if (lookup_registration_code == '') {
+                registration_code_status_form_error.attr('style', 'display: block !important');
+                registration_code_status_form_error.text(gettext('Please enter the Enrollment Code.'));
+                lookup_button.removeAttr('disabled');
+                return false;
+            }
+            $.ajax({
+                type: "POST",
+                data: {
+                    "registration_code"  : lookup_registration_code
+                },
+                url: url,
+                success: function (data) {
+                    lookup_button.removeAttr('disabled');
+                    if (data.is_registration_code_exists == 'false') {
+                        registration_code_status_form_error.attr('style', 'display: none');
+                        registration_code_status_form_error.attr('style', 'display: block !important');
+                        registration_code_status_form_error.text(gettext(data.message));
+                    }
+                    else {
+                        var is_registration_code_valid = data.is_registration_code_valid.toString().capitalizeFirstLetter();
```

Comment:
`data.is_registration_code_valid` and `data.is_registration_code_redeemed` should already be booleans. There is no need to convert them to strings. Even if you continue converting them, it would be far simpler to call [`String. toLowerCase()`](http://www.w3schools.com/jsref/jsref_tolowercase.asp) instead of creating `capitalizeFirstLetter()`.